src/utils/tokenMetadata.ts

```diff
@@ -36,12 +36,13 @@ function createGetMultiAssetTxMintMetadataQuery(assets: PolicyIdAssetMapType) {
   const whereConditions = Object.keys(assets)
     .map((policyIdHex: string) => {
       const assetNames = assets?.[policyIdHex] ?? [];
+      const policID = `( encode(ma.policy, 'hex') = ($${index++})::varchar and`;
       const query = assetNames
-        .map((_) => 
-            `(encode(ma.name, 'hex') = ($${index++})::varchar
-          and encode(ma.policy, 'hex') = ($${index++})::varchar )`)
+        .map((_) => `encode(ma.name, 'hex') = ($${index++})::varchar`)
```

Comment:
If this approach is faster, maybe we should do the same thing for `utxoForTransaction`? (match on hash and then OR for the index)

Reply:
i refactored it but it didnt make much difference to that pr, since there are already limited number of index per txHash